x-pack/plugins/security_solution/public/common/hooks/eql/api.ts

```diff
@@ -23,6 +24,7 @@ interface Params {
   data: DataPublicPluginStart;
   signal: AbortSignal;
   runtimeMappings: estypes.MappingRuntimeFields | undefined;
+  options: Omit<EqlOptionsSelected, 'query' | 'size'>;
```

Comment:
Should this one be optional?
```suggestion
  options?: Omit<EqlOptionsSelected, 'query' | 'size'>;
```

Reply:
I made it required to begin in order to light up all the call sites with typescript. As with `runtimeMappings` I tend to prefer arguments to be required (but allow `undefined` if it makes sense) so I'd likely do the same here (although `options: {}` also satisfies). Since there's just the single call currently, I'm leaning toward explicitness over developer ergonomics.

Reply:
It makes complete sense, but since it is called "_options_", it denotes they are values that can be optionally defined, or just omitted (that's not the case for `runtimeMappings`). So, in this specific case, I would lean towards making it optional instead of forcing the "options" to be always specified.

Nevertheless, that's just a subtility (pretty subjective probably), so I think we are perfectly good anyway with `options: Omit<EqlOptionsSelected, 'query' | 'size'> | undefined;` 👍

Reply:
Ahh, yes, I see your point now! "Options" does imply "optionality" 😉

Reply:
e837f19373ff9f7b99dcc6825878f62b055afc1a